packages/webapp/src/components/PrivateRoute.tsx

```diff
@@ -1,19 +1,33 @@
 import { Outlet, Navigate } from 'react-router-dom';
 import { useMeta } from '../hooks/useMeta';
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
 import { useStore } from '../store';
 import { useAnalyticsIdentify } from '../utils/analytics';
 import { useUser } from '../hooks/useUser';
+import PageNotFound from '../pages/PageNotFound';
 
-const PrivateRoute: React.FC = () => {
+export const PrivateRoute: React.FC = () => {
     const { meta, error, loading } = useMeta();
-    const { user } = useUser();
+    const [notFoundEnv, setNotFoundEnv] = useState(false);
+    const [ready, setReady] = useState(false);
+    const { user } = useUser(Boolean(meta && ready && !notFoundEnv));
     const identify = useAnalyticsIdentify();
 
+    const env = useStore((state) => state.cookieValue);
     const setStoredEnvs = useStore((state) => state.setEnvs);
     const setBaseUrl = useStore((state) => state.setBaseUrl);
     const setEmail = useStore((state) => state.setEmail);
     const setDebugMode = useStore((state) => state.setDebugMode);
+    const setCookieValue = useStore((state) => state.setCookieValue);
+
+    useEffect(() => {
+        // sync path with cookie
+        const pathSplit = location.pathname.split('/');
+        if (env !== pathSplit[1]) {
```

Comment:
logically I know that there will always be an element at index position 1, but this kind of access still makes me nervous

Reply:
hehe yes I can add some safety